tests/monitors/telegraf_snmp/snmp_test.py

```diff
@@ -0,0 +1,36 @@
+from functools import partial as p
+import pytest
+import string
+
+from tests.helpers.util import wait_for, run_agent, run_container, container_ip
+from tests.helpers.assertions import has_datapoint_with_dim
+
+pytestmark = [pytest.mark.telegraf, pytest.mark.snmp, pytest.mark.monitor_with_endpoints]
+
+monitor_config = string.Template("""
+monitors:
+- type: telegraf/snmp
+  agents:
+    - "$host:161"
+  version: 2
+  community: "public"
+  fields:
+    - name: "uptime"
+      oid: ".1.3.6.1.2.1.1.3.0"
+""")
+
+
+def test_snmp():
+    # snmp-simulator source is available at: https://github.com/xeemetric/snmp-simulator
+    # the hard coded uptime OID used in this test was fetched using the following command
+    # $ snmpget -On -v2c -c public <snmp simulator host>:161 system.sysUpTime.0
+    with run_container("xeemetric/snmp-simulator") as test_container:
+        host = container_ip(test_container)
+        config = monitor_config.substitute(host=host)
+
+        # TODO: write a better check for an open udp port
+        import time
+        time.sleep(10)
```

Comment:
Unless the monitor is incapable of handling the SNMP service coming up after the monitor has started (which is a pretty bad implemetation), this wait is unnecessary.